packages/core/src/browser/preferences/preference-service.spec.ts

```diff
@@ -0,0 +1,459 @@
+            assert.ok(!preferences.has('[json].editor.tabSize'));
+
+            schema.registerOverrideIdentifier('json');
+
+            assert.deepStrictEqual(expected, preferences.inspect('editor.tabSize'));
+            assert.deepStrictEqual({
+                ...expected,
+                preferenceName: '[json].editor.tabSize'
+            }, preferences.inspect('[json].editor.tabSize'));
+        });
+
+        it('inspect #2', () => {
+            const { preferences, schema } = prepareServices();
+
+            const expected = {
+                preferenceName: 'editor.tabSize',
+                defaultValue: 4,
+                globalValue: undefined,
+                workspaceValue: undefined,
+                workspaceFolderValue: undefined,
+            };
+            assert.deepStrictEqual(expected, preferences.inspect('editor.tabSize'));
+            assert.ok(!preferences.has('[json].editor.tabSize'));
+
+            schema.registerOverrideIdentifier('json');
+            preferences.set('[json].editor.tabSize', 2, PreferenceScope.User);
+
+            assert.deepStrictEqual(expected, preferences.inspect('editor.tabSize'));
+            assert.deepStrictEqual({
+                ...expected,
+                preferenceName: '[json].editor.tabSize',
+                globalValue: 2
+            }, preferences.inspect('[json].editor.tabSize'));
+        });
+
+        it('onPreferenceChanged #0', () => {
+            const { preferences, schema } = prepareServices();
+
+            const events: PreferenceChange[] = [];
+            preferences.onPreferenceChanged(event => events.push(event));
+
+            schema.registerOverrideIdentifier('json');
+            preferences.set('[json].editor.tabSize', 2, PreferenceScope.User);
+            preferences.set('editor.tabSize', 3, PreferenceScope.User);
+
+            assert.deepStrictEqual([{
+                preferenceName: '[json].editor.tabSize',
+                newValue: 2
+            }, {
+                preferenceName: 'editor.tabSize',
+                newValue: 3
+            }], events.map(e => ({
+                preferenceName: e.preferenceName,
+                newValue: e.newValue
+            })));
+        });
+
+        it('onPreferenceChanged #1', () => {
+            const { preferences, schema } = prepareServices();
+
+            const events: PreferenceChange[] = [];
+            preferences.onPreferenceChanged(event => events.push(event));
+
+            schema.registerOverrideIdentifier('json');
+            preferences.set('editor.tabSize', 2, PreferenceScope.User);
+
+            assert.deepStrictEqual([{
+                preferenceName: 'editor.tabSize',
+                newValue: 2
+            }, {
+                preferenceName: '[json].editor.tabSize',
+                newValue: 2
+            }], events.map(e => ({
+                preferenceName: e.preferenceName,
+                newValue: e.newValue
+            })));
+        });
```

Comment:
there are irrelevant events for overrides
```ts
        it('onPreferenceChanged #2', () => {
            const { preferences, schema } = prepareServices();

            schema.registerOverrideIdentifier('json');
            schema.registerOverrideIdentifier('javascript');
            preferences.set('[json].editor.tabSize', 2, PreferenceScope.User);
            preferences.set('editor.tabSize', 3, PreferenceScope.User);

            const events: PreferenceChange[] = [];
            preferences.getPreferences(undefined, 'json').onPreferenceChanged((event: PreferenceChange) => events.push(event));

            preferences.set('[javascript].editor.tabSize', 4, PreferenceScope.User);

            assert.deepStrictEqual([], events.map(e => ({
                preferenceName: e.preferenceName,
                newValue: e.newValue
            })), 'changes not relevant to json override should be ignored');
        });
```

Reply:
added that testcase